Makefile.in

```diff
@@ -10,6 +10,14 @@
 
 srcdir = @srcdir@
 VPATH = @srcdir@
+builddir = @builddir@
+
+# when srcdir and builddir are different we need to get config.h from builddir.
```

Comment:
A common convention is to require the src TLD to be the CWD when running the configure and the make. All references to internal headers might be like in jeeps/blah.cc #include "../config.h" and garmin.cc #include "config.h"

This is probably more robust than most, I suspect...

Reply:
The reference to internal headers can still work as you cited, even if the build is done somewhere besides the source tree, i.e. they will be resolved relative to the source file that has the include.